### PR TITLE
Updates Engine to use new ruleset structure

### PR DESCRIPTION
**Pull request by JavierBejMen**
This PR merges the intermediate branch that updates the Engine to the new ruleset structure into master branch.

### CODE REVIEW

**Review by juliancnn — APPROVED** [review]
LGTM!

**Review by NahuFigueroa97 — APPROVED** [review]
LGTM!

**Review by JcabreraC — APPROVED** [review]
LGTM !

File: src/engine/source/README.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+# Directory structure
+
+```plaintext
+source/
+│
+├── api/
+├── base/
+├── bk/
+├── builder/
```
Comment (Dwordcito) [inline src/engine/source/README.md:14]: missing indexerconnector, vdscanner, feedmanager and apiserver, is it really necessary to have this? I don't think we will add many modules, but it is something easy to become obsolete and forget.

File: src/engine/source/README.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+└── test/
+    ├── mocks/
+    │   └── <component>_mocks.h 
+    │
+    └── src/
+        ├── component/
+        │   └── <component>_test.cpp
+        │
+        └── unitario/
+            └── <component>_unit_test.cpp
```
Comment (Dwordcito) [inline src/engine/source/README.md:63]:
```suggestion
            └── <class>_test.cpp
```

**Review by Dwordcito — CHANGES_REQUESTED** [review]
Minor things

**Comment by JcabreraC** [comment]
All checks are passed before the README is deleted:
imagen

**Review by JcabreraC — APPROVED** [review]
LGTM !

**Review by Dwordcito — APPROVED** [review]

**Pull request merged**